lib/util/stream_utils.js

```diff
@@ -532,10 +531,10 @@ shaka.util.StreamUtils = class {
           variant, usePersistentLicenses, srcEquals);
 
       for (const config of decodingConfigs) {
-        operations.push(getVariantDecodingInfos(variant, config));
+        // eslint-disable-next-line no-await-in-loop
+        await getVariantDecodingInfos(variant, config);
```

Comment:
The reason eslint has a check for this is because serializing these async calls is bound to be slower than doing them in parallel.  Why did you change this?

Reply:
The reason for this change is the result of testing across a couple of devices it yielded better performance. Across browsers, TVs and Chromecast. I saw no real time improvement between between `Promise.all` and `await` in a loop as `requestMediaKeySystemAccess` appears to not benefit from a multithreaded approach. 

I performed these tests with one of our assets that has many variants.
```
Browser
(promise.all)
56ms-75ms async
73ms-84ms cache + async
60ms-74ms cache + pre-cache async

(awaiting in a loop)
58ms -71ms sync 
40ms-54ms cache sync
35ms-46ms cache + pre-cache sync

Vizio TV
(promise.all)
4665 - 5940 async

Async + cache
4288 - 4576 (**This is the time on a secondary load as the initial 
doesn't benefit from the cache as all threads appear to stack and 
wait for `requestMediaKeySystemAccess`  calls to resolve synchronously**)

Sync
4594 - 5753

Sync + cache
3445 - 3615 (** gains benefit from the caching on initial play throughs **) 
```

I am happy to revert this part if you see different results, as I appreciate you will be aware of more use cases and devices.

Reply:
Are we sure there's a benefit in running requestMediaKeySystemAccess in parallel ?
I would agree to parallelise it, if it was network bound, but initialising the CDM is probably CPU bound instead and potentially even protected from running in multiple threads... which would explain the results we see above.

Reply:
Ah, okay.  It's possible that there is some startup state that has to be computed, and serializing queries may allow that state to be reused in subsequent queries.  Imagine this:

```
  Query A        Query B        Query C
     |              |              |
  Slow           Slow           Slow
  platform       platform       platform
  check          check          check
(cache miss)   (cache miss)   (cache miss)
```

vs this:

```
  Query A
     |
  Slow
  platform
  check
(cache miss)

  Query B
     |
  Cached
  platform
  check

  Query C
     |
  Cached
  platform
  check
```

Purely hypothetical.  I have no idea how those queries are handled by the device.

Reply:
I have added the requested comment.